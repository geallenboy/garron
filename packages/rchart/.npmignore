.umi
es/.umi
cjs/.umi
types
tests
public
src
.umirc
babel.config.js
jest.config.js
tsconfig.json