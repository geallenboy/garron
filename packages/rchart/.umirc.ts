import { defineConfig } from 'dumi';
export default defineConfig({
  title: 'react图表',
  favicon: './images/logo.svg',
  logo: './images/logo.svg',
  resolve: {
    includes: ['docs']
  },
  mode: 'site',
  menus: {
    '/': [
      {
        title: '首页',
        path: 'index'
      }
    ],
    '/guide': [
      {
        title: '介绍',
        path: '/guide'
      }
    ],
    '/plots': [
      {
        title: '折线图',
        children: [
          '/plots/line/api',
          '/plots/line/base1',
          '/plots/line/base2',
          '/plots/line/base3',
          '/plots/line/base4'
        ]
      },
      {
        title: '面积图',
        children: ['/plots/area/api', '/plots/area/base1', '/plots/area/base2', '/plots/area/base3']
      },
      {
        title: '柱状图',
        children: [
          '/plots/column/api',
          '/plots/column/base1',
          '/plots/column/base2',
          '/plots/column/base3'
        ]
      },
      {
        title: '条形图',
        children: [
          '/plots/bar/api',
          '/plots/bar/base1',
          '/plots/bar/base2',
          '/plots/bar/base3',
          '/plots/bar/base4'
        ]
      },
      {
        title: '饼图',
        children: ['/plots/pie/api', '/plots/pie/base1', '/plots/pie/base2']
      },
      {
        title: '散点图',
        children: ['/plots/scatter/api', '/plots/scatter/base1', '/plots/scatter/base2']
      },
      {
        title: '玫瑰图',
        children: ['/plots/rose/api', '/plots/rose/base1']
      },
      {
        title: '关系图',
        children: ['/plots/relation/api', '/plots/relation/base1', '/plots/relation/base2']
      }
    ],
    '/maps': [
      {
        title: '区域地图',
        children: ['/maps/area/api', '/maps/area/base1']
      }
    ]
  },
  navs: [
    {
      title: '指南',
      path: '/guide'
    },
    {
      title: '图表',
      path: '/plots'
    },
    {
      title: '地图',
      path: '/maps'
    },
    {
      title: 'GitHub',
      path: 'https://github.com/geallenboy/garron'
    }
  ]
});
