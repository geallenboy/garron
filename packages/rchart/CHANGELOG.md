# @garron/rchart

## 2.0.0

### Major Changes

- 引入 changeset 工具
