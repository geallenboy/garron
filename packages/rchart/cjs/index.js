'use strict';

var _interopRequireDefault = require('@babel/runtime/helpers/interopRequireDefault');

Object.defineProperty(exports, '__esModule', {
  value: true
});
Object.defineProperty(exports, 'AreaChart', {
  enumerable: true,
  get: function () {
    return _area.default;
  }
});
Object.defineProperty(exports, 'AreaMap', {
  enumerable: true,
  get: function () {
    return _area2.default;
  }
});
Object.defineProperty(exports, 'BarChart', {
  enumerable: true,
  get: function () {
    return _bar.default;
  }
});
Object.defineProperty(exports, 'BidirectionalBarChart', {
  enumerable: true,
  get: function () {
    return _bidirectionalBar.default;
  }
});
Object.defineProperty(exports, 'BoxChart', {
  enumerable: true,
  get: function () {
    return _box.default;
  }
});
Object.defineProperty(exports, 'BulletChart', {
  enumerable: true,
  get: function () {
    return _bullet.default;
  }
});
Object.defineProperty(exports, 'ChordChart', {
  enumerable: true,
  get: function () {
    return _chord.default;
  }
});
Object.defineProperty(exports, 'ColumnChart', {
  enumerable: true,
  get: function () {
    return _column.default;
  }
});
Object.defineProperty(exports, 'DualAxesChart', {
  enumerable: true,
  get: function () {
    return _dualAxes.default;
  }
});
Object.defineProperty(exports, 'FunnelChart', {
  enumerable: true,
  get: function () {
    return _funnel.default;
  }
});
Object.defineProperty(exports, 'G2', {
  enumerable: true,
  get: function () {
    return _g2plot.G2;
  }
});
Object.defineProperty(exports, 'GaugeChart', {
  enumerable: true,
  get: function () {
    return _gauge.default;
  }
});
Object.defineProperty(exports, 'HeatmapChart', {
  enumerable: true,
  get: function () {
    return _heatmap.default;
  }
});
Object.defineProperty(exports, 'LineChart', {
  enumerable: true,
  get: function () {
    return _line.default;
  }
});
Object.defineProperty(exports, 'LiquidChart', {
  enumerable: true,
  get: function () {
    return _liquid.default;
  }
});
Object.defineProperty(exports, 'PieChart', {
  enumerable: true,
  get: function () {
    return _pie.default;
  }
});
Object.defineProperty(exports, 'RadarChart', {
  enumerable: true,
  get: function () {
    return _radar.default;
  }
});
Object.defineProperty(exports, 'RadialBarChart', {
  enumerable: true,
  get: function () {
    return _radialBar.default;
  }
});
Object.defineProperty(exports, 'RoseChart', {
  enumerable: true,
  get: function () {
    return _rose.default;
  }
});
Object.defineProperty(exports, 'SankeyChart', {
  enumerable: true,
  get: function () {
    return _sankey.default;
  }
});
Object.defineProperty(exports, 'ScatterChart', {
  enumerable: true,
  get: function () {
    return _scatter.default;
  }
});
Object.defineProperty(exports, 'StockChart', {
  enumerable: true,
  get: function () {
    return _stock.default;
  }
});
Object.defineProperty(exports, 'SunburstChart', {
  enumerable: true,
  get: function () {
    return _sunburst.default;
  }
});
Object.defineProperty(exports, 'TreemapChart', {
  enumerable: true,
  get: function () {
    return _treemap.default;
  }
});
Object.defineProperty(exports, 'WaterfallChart', {
  enumerable: true,
  get: function () {
    return _waterfall.default;
  }
});
Object.defineProperty(exports, 'WordCloudChart', {
  enumerable: true,
  get: function () {
    return _wordCloud.default;
  }
});
Object.defineProperty(exports, 'measureTextWidth', {
  enumerable: true,
  get: function () {
    return _g2plot.measureTextWidth;
  }
});

var _g2plot = require('@antv/g2plot');

var _area = _interopRequireDefault(require('./plots/area'));

var _bar = _interopRequireDefault(require('./plots/bar'));

var _bidirectionalBar = _interopRequireDefault(require('./plots/bidirectional-bar'));

var _box = _interopRequireDefault(require('./plots/box'));

var _bullet = _interopRequireDefault(require('./plots/bullet'));

var _chord = _interopRequireDefault(require('./plots/chord'));

var _column = _interopRequireDefault(require('./plots/column'));

var _dualAxes = _interopRequireDefault(require('./plots/dual-axes'));

var _funnel = _interopRequireDefault(require('./plots/funnel'));

var _gauge = _interopRequireDefault(require('./plots/gauge'));

var _heatmap = _interopRequireDefault(require('./plots/heatmap'));

var _line = _interopRequireDefault(require('./plots/line'));

var _liquid = _interopRequireDefault(require('./plots/liquid'));

var _pie = _interopRequireDefault(require('./plots/pie'));

var _radar = _interopRequireDefault(require('./plots/radar'));

var _radialBar = _interopRequireDefault(require('./plots/radial-bar'));

var _rose = _interopRequireDefault(require('./plots/rose'));

var _sankey = _interopRequireDefault(require('./plots/sankey'));

var _scatter = _interopRequireDefault(require('./plots/scatter'));

var _stock = _interopRequireDefault(require('./plots/stock'));

var _sunburst = _interopRequireDefault(require('./plots/sunburst'));

var _treemap = _interopRequireDefault(require('./plots/treemap'));

var _waterfall = _interopRequireDefault(require('./plots/waterfall'));

var _wordCloud = _interopRequireDefault(require('./plots/word-cloud'));

var _area2 = _interopRequireDefault(require('./maps/area'));
