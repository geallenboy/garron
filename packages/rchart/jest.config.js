module.exports = {
  setupFiles: ['jest-canvas-mock', './tests/setups/worker.js', './tests/setups/url.js'],
  transformIgnorePatterns: ['/node_modules/', '/src/.umi/'],
  collectCoverageFrom: ['./src/**/*.{ts,tsx}', '!./**/*.d.ts', '!src/.umi/**/*.{ts,tsx}'],
  coverageReporters: ['json', 'text', 'lcov', 'clover', 'cobertura'],
  testEnvironment: 'jsdom'
};
