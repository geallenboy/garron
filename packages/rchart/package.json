{
  "name": "@garron/rchart",
  "version": "1.0.5",
  "description": "react chart",
  "keywords": [
    "chart",
    "G2Plot",
    "g2"
  ],
  "license": "MIT",
  "main": "cjs/index.js",
  "module": "es/index.js",
  "directories": {
    "cjs": "cjs",
    "es": "es"
  },
  "files": [
    "cjs",
    "es",
    "typings",
    "*.d.ts"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "npm run build:es & npm run build:commonjs",
    "build:commonjs": "cross-env BABEL_ENV=commonjs babel src -d cjs --extensions .ts,.tsx --delete-dir-on-start",
    "build:docs": "dumi build",
    "build:es": "cross-env BABEL_ENV=es babel src -d es --extensions .ts,.tsx --delete-dir-on-start",
    "build:types": "tsc --declarationDir ./cjs & tsc --declarationDir ./es",
    "dev": "dumi dev",
    "test": "jest -u",
    "test-cov": "jest -u --coverage"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not dead"
  ],
  "dependencies": {
    "@antv/g2plot": "^2.3.24",
    "@antv/l7plot": "^0.1.0",
    "jest-environment-jsdom": "^29.0.2",
    "react-content-loader": "^6.2.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.18.10",
    "@babel/plugin-proposal-class-properties": "^7.18.6",
    "@babel/plugin-proposal-private-property-in-object": "^7.18.6",
    "@babel/plugin-transform-runtime": "^7.18.10",
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@types/jest": "^29.0.0",
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "@types/react-test-renderer": "^18.0.0",
    "@types/string-template": "^1.0.2",
    "@umijs/plugin-analytics": "^0.2.3",
    "cross-env": "^7.0.3",
    "dumi": "^1.1.40",
    "jest": "^29.0.2",
    "jest-canvas-mock": "^2.4.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-test-renderer": "^18.2.0",
    "string-template": "^1.0.0",
    "ts-jest": "^28.0.8",
    "ts-node": "^10.9.1",
    "typescript": "^4.5.2"
  }
}
