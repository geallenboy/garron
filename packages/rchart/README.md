## å®‰è£…

```bash
npm install @garron/rchart
```

## ä½¿ç”¨

```js
import { LineChart } from '@garron/rchart';
```

## Examples

```tsx
import React, { useCallback } from 'react';
import { LineChart, ColumnChart, PieChart, BarChart } from '@garron/rchart';

const config = {
  height: 350,
  autoFit: true,
  xField: 'year',
  yField: 'value',
  smooth: true,
  meta: {
    value: {
      max: 15
    }
  },
  data: [
    { year: '1991', value: 3 },
    { year: '1992', value: 4 },
    { year: '1993', value: 3.5 },
    { year: '1994', value: 5 },
    { year: '1995', value: 4.9 },
    { year: '1996', value: 6 },
    { year: '1997', value: 7 },
    { year: '1998', value: 9 },
    { year: '1999', value: 11 }
  ]
};

const pieConfig = {
  height: 350,
  autoFit: true,
  padding: 'auto',
  radius: 1,
  data: [
    {
      type: 'A',
      value: 27
    },
    {
      type: 'B',
      value: 25
    },
    {
      type: 'C',
      value: 18
    },
    {
      type: 'D',
      value: 15
    },
    {
      type: 'E',
      value: 10
    },
    {
      type: 'Others',
      value: 5
    }
  ],
  angleField: 'value',
  colorField: 'type',
  label: {
    visible: true,
    type: 'inner'
  },
  legend: {
    visible: false
  }
};

export default () => {
  return (
    <div
      style={{
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-between'
      }}
    >
      <LineChart
        style={{
          width: '48%',
          minWidth: 350,
          marginTop: 32
        }}
        {...config}
      />
      <ColumnChart
        style={{
          width: '48%',
          minWidth: 350,
          marginTop: 32
        }}
        {...config}
      />
      <BarChart
        style={{
          width: '48%',
          minWidth: 350,
          marginTop: 32
        }}
        {...config}
        xField="value"
        yField="year"
      />
      <PieChart
        style={{
          width: '48%',
          minWidth: 350,
          marginTop: 32
        }}
        {...pieConfig}
      />
    </div>
  )
}

```
