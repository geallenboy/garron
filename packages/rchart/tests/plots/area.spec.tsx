import React from 'react';
import { create } from 'react-test-renderer';

import { AreaChart } from '@garron/rchart';

describe('AreaChart', () => {
  test('渲染AreaChart测试', () => {
    const renderer = create(<AreaChart data={[]} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
