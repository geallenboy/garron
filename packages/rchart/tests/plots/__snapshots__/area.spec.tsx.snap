// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaChart 渲染AreaChart测试 1`] = `<div />`;
